helix-core/src/test/java/org/apache/helix/cloud/virtualTopologyGroup/TestVirtualTopologyGroupAssignment.java

```diff
@@ -0,0 +1,160 @@
+
+  @BeforeTest
+  public void prepare() {
+    _zoneMapping = new HashMap<>();
+    _zoneMapping.put("c", Sets.newHashSet("9", "8", "7"));
+    _zoneMapping.put("a", Sets.newHashSet("2", "3", "1"));
+    _zoneMapping.put("z", Sets.newHashSet("b", "c", "d", "a"));
+    _zoneMapping.put("b", Sets.newHashSet("5", "4", "6"));
+  }
+
+  @Test
+  public void testFlattenZoneMapping() {
+    Assert.assertEquals(HelixUtil.sortAndFlattenZoneMapping(_zoneMapping), _flattenExpected);
+  }
+
+  @Test(dataProvider = "getMappingTests")
+  public void testAssignmentScheme(int numGroups, Map<String, Set<String>> expected,
+      VirtualGroupAssignmentAlgorithm algorithm) {
+    Assert.assertEquals(algorithm.computeAssignment(numGroups, GROUP_NAME, _zoneMapping), expected);
+  }
+
+  @Test
+  public void assignmentBenchmark() {
+    Map<String, Set<String>> testMapping = createZoneMapping(new int[] {40, 40, 40, 40, 40}, "zone");
+    singleAssignmentBenchmark(testMapping, 13);
+    singleAssignmentBenchmark(testMapping, 20);
+
+    int numGroups = 20;
+    Map<String, Set<String>> testMapping2 = createZoneMapping(new int[] {40, 40, 40, 40, 40}, "zone");
+    testMapping2.get("zone_2").add("2_201");
+    testMapping2.get("zone_2").add("2_202");
+    testMapping2.get("zone_3").add("3_301");
+    testMapping2.get("zone_3").add("3_302");
+    compareAssignmentUnderDifferentZoneMapping(testMapping, testMapping2, numGroups, VirtualTopologyGroupScheme.FIFO);
+
+    Map<String, Set<String>> testMapping3 = createZoneMapping(new int[] {40, 40, 40, 40, 40}, "zone");
+    testMapping3.get("zone_2").remove("2_13");
+    testMapping3.get("zone_2").remove("2_8");
+    testMapping3.get("zone_4").remove("4_8");
+    testMapping3.get("zone_4").remove("4_17");
+    compareAssignmentUnderDifferentZoneMapping(testMapping, testMapping3, numGroups, VirtualTopologyGroupScheme.FIFO);
+  }
+
+  private static void singleAssignmentBenchmark(Map<String, Set<String>> testMapping, int numGroups) {
+    System.out.println("Test mapping with numGroups: " + numGroups);
+    System.out.println(testMapping);
+    System.out.println("==========");
+    validate(numGroups, testMapping, VirtualTopologyGroupScheme.FIFO);
+  }
+
+  private static void compareAssignmentUnderDifferentZoneMapping(
+      Map<String, Set<String>> baseMapping,
+      Map<String, Set<String>> testMapping,
+      int numGroups,
+      VirtualGroupAssignmentAlgorithm algorithm) {
+    Map<String, Set<String>> baseAssignment = algorithm.computeAssignment(numGroups, GROUP_NAME, baseMapping);
+    Map<String, Set<String>> testAssignment = algorithm.computeAssignment(numGroups, GROUP_NAME, testMapping);
+    AssignmentEvaluation base = new AssignmentEvaluation(baseAssignment, baseMapping);
+    AssignmentEvaluation test = new AssignmentEvaluation(testAssignment, testMapping);
+    System.out.println("Diff for " + algorithm + " : " + base.compareAssignments(test));
+  }
```

Comment:
Why do we need to include this comparison in our test suite? Since your algorithm is deterministic, the result won't change no matter what. And I don't see the point of keeping running it for each commit. What is the value of knowing the diff between base and test case? The test does not really validate/assert anything.

Reply:
To clarify, it's for computing the stability of the assignment under cluster change (add or remove instances). It's part of the benchmark, so it's not validating anything, but compute and print statistics for reference.
I do see some value for the benchmark utils, but if it's not the coding convention, I'm totally ok to remove.

Reply:
So what's the expectation for people that run the test suite? Do they need to check the detail in the log to make sure the test runs correctly? Then it does not make much sense. If you can make it pass or fail with a standard, then you can keep it.